libdevice/include/assert.h

```diff
@@ -0,0 +1,76 @@
+//==--------- devicelib-assert.h - wrapper definition for C assert ---------==//
```

Comment:
Another option is move this header to the compiler headers.
Here the same wrapper implementation for PTX target: https://github.com/intel/llvm/blob/sycl/clang/lib/Headers/__clang_cuda_runtime_wrapper.h#L335.